subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[toolchains]]
+= Toolchains for JVM projects
+
+By default, Gradle uses the same Java version for running Gradle itself and building JVM projects.
+
+This is not always desirable.
+Building projects with different Java versions on different developer machines and CI servers may lead to unexpected issues Additionally, you may want to build a project using a Java version that running Gradle is not compatible with.
+
+A Java Toolchain (from now on referred to simply as toolchain) is a set of tools, usually taken from a local JRE/JDK installation that are used to configure different aspects of a build.
+Compile tasks may use `javac` as their compiler, test and exec tasks may use the `java` command while
+`javadoc` will be used to generate documentation.
+
+== Consuming Toolchains
+
+A build can globally define what toolchain it targets by stating the Java Language version it needs:
+
+====
+include::sample[dir="samples/java/jvm-multi-project-with-toolchains/groovy/",files="buildSrc/src/main/groovy/myproject.java-conventions.gradle[tags=toolchain]"]
+include::sample[dir="samples/java/jvm-multi-project-with-toolchains/kotlin/",files="buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts[tags=toolchain]"]
+====
+
+Executing the build (e.g. using `gradle check`) will now handle several things for you and others running your build
+
+1. Setup all compile and test tasks to use the defined toolchain and not the JVM that Gradle is running with
+2. Gradle detects <<sec:auto_detection,locally installed JVMs>>
+3. Gradle chooses a JRE/JDK matching the requirements of the build (in this case a JVM supporting Java 14)
+4. If no matching JVM is found, it will automatically download a matching JDK from https://adoptopenjdk.net/[AdoptOpenJDK]
+
+== Specify custom toolchains for individual tasks
+
+In case you want to tweak which toolchain is used for a specific task, you can specify the exact tool a task is using.
+For example, a `Test` task exposes a `JavaLauncher` property that defines which java executable to use to launch the tests.
+
+In the example below, we configure the default toolchain to use JDK8 for compiling and testing.
+Additionally, we introduce a new Test task that is going to run our unit tests but using a JDK 14. Depending on the task, a JRE might be enough while for other tasks (e.g. compilation), a JDK is required.
```

Comment:
Missing line break on "."

Reply:
I would also move the code extract after this first sentence and have the rest of the text below it.